src/metrics/include/metrics/MetricsReporter.hpp

```diff
@@ -235,6 +242,75 @@ class ReporterT final {
         }
     }
 
+    void reportCedarCsv(std::ostream& out,
+                        long long systemTime,
+                        long long metricsTime,
+                        v1::Permission perm) const {
+        out << "Clocks" << std::endl;
+        out << "clock,nanoseconds" << std::endl;
+        writeClocks(out, systemTime, metricsTime);
+        out << std::endl;
+
+        // We use an ordered map here to avoid defining a custom hash function for
+        // std::pair<std::string, std::string>. There aren't likely to be many (Actor, Operation)
+        // combinations for this to matter too much in terms of efficiency.
+        auto opThreadCounts = std::map<std::pair<std::string, std::string>, size_t>{};
+        out << "OperationThreadCounts" << std::endl;
+        out << "actor,operation,workers" << std::endl;
+        for (const auto& [actorName, opsByType] : _registry->getOps(perm)) {
+            for (const auto& [opName, opsByThread] : opsByType) {
+                if (actorName == "Genny" &&
+                    (opName == "ActorStarted" || opName == "ActorFinished")) {
+                    // The cedar-csv metrics format ignores the Genny.ActorStarted and
+                    // Genny.ActorFinished operations reported by the DefaultDriver because the
+                    // OperationThreadCounts section effectively tracks the number of concurrent
+                    // actors and that number isn't meaningfully changing over time.
+                    continue;
+                }
+
+                opThreadCounts[std::make_pair(actorName, opName)] += opsByThread.size();
```

Comment:
any reason to use a map versus a [multiset](http://www.cplusplus.com/reference/unordered_set/unordered_multiset/)? I prefer the map version but some may argue multiset is more appropriate.

Reply:
I hadn't looked at the multiset type in C++ before. I think because we want the count of all elements and not just the count for a specific key that the `map<std::pair<std::string, std::string>, size_t>` presents an appropriate interface.